src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+            [$scriptId],
+            -1, 
+            "rsync long arguments", 
+            "rsync short arguments", 
+            "ssh arguments", 
+            "root@172.0.0.1"
+        );
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+    public function testCreateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withMaxParallelJobs($clientName, -1);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+
+    public function testCreateClientInvalidName(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertHydraError("An exception occurred while executing 'INSERT INTO Client (description, isActive, name, url, quota, sshArgs, rsyncShortArgs, rsyncLongArgs, state, maxParallelJobs, data, owner_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [null, 1, \"".$clientName."\", \"\", -1, null, null, null, \"NOT READY\", 1, null, 1]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '".$clientName."' for key 'Client.UNIQ_C0E801635E237E06'");
+    }
+
+    public function testCreateClientInvalidOwner(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withOwner($clientName, self::UNEXISTING_ID);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('Incorrect owner id');
+    }
+
+    public function testCreateClientInvalidPostScript(): void
+    {
+        $httpClient = $this->authenticate();
+        $scriptId = $this->getScriptId($httpClient, 'script_not_client_post');
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withPostScripts($clientName, [$scriptId]);
```

Comment:
If you really don't care what scriptId's to use why not just use
`ClientMother::withPostScripts()` and let it choose what postscripts it sets.

Of course this is something general, not just por this withPostScripts but for withPreScripts also. Maybe you could make the scripts argument optional so that this method suffixes for scripts set automatically by the mother and script that you have to set explicitly.